### PR TITLE
Fix GetMarketReport predefined period all

### LINKED ISSUE #605: PREDEFINED_PERIOD_ALL is stoping at previous calendar year
All 2022 data points are missing.

### PR DESCRIPTION
**Pull request by sekulicd**
This fixes bug that was happening when GetMarketReport RPC was invoked with predefined period PREDEFINED_PERIOD_ALL.

This closes #605 

@tiero @altafan @Janaka-Steph please review.

### CODE REVIEW

**Comment by altafan** [comment]
@sekulicd please add a test that demonstrates the bug is fixed.

**Comment by sekulicd** [comment]
> @sekulicd can you please add a test that demonstrates the bug is fixed?

Added test for application layer GetMarketReport

File: internal/core/application/types.go [hunk 1]
```diff
@@ -542,7 +542,7 @@ func (t *TimeRange) Validate() error {
 type PredefinedPeriod int
 
 func (p *PredefinedPeriod) validate() error {
-	if *p > LastYear {
+	if *p > All {
 		return fmt.Errorf("PredefinedPeriod cant be > %v", LastYear)
```
Comment (altafan) [inline internal/core/application/types.go:546]:
```suggestion
		return fmt.Errorf("PredefinedPeriod cant be > %v", All)
```

Comment (sekulicd) [reply internal/core/application/types.go]: Done

**Review by tiero — APPROVED** [review]

**Pull request merged**